### PR TITLE
stdlib: Add unavailable declarations for APIs renamed in SE-0118

### PR DESCRIPTION
**Pull request by rintaro**
#### What's in this pull request?

[SE-0118](https://github.com/apple/swift-evolution/blob/master/proposals/0118-closure-parameter-names-and-labels.md) revised several parameter labels.

"incorrect argument labels" diagnosis can catch most of them, but not always. e.g. [SR-2318](https://bugs.swift.org/browse/SR-2318)

This PR adds explicit `unavailable, renamed` declarations for them.

`Sequence.sort(_:)` => `sorted(by:)`
`Sequence.sort()` => `sorted()`
`Sequence.elementsEqual(_:isEquivalent)` => `elementsEqual(_:by:)`
`Sequence.contains(_:)` => `contains(where:)`
`Sequence.reduce(_:combine:)` => `reduce(_:_:)`
`UnicodeCodec.encode(_:output:)` => `encode(_:into:)`
`ManagedBuffer.create(_:initialValue:)` => `create(minimumCapacity:makingHeaderWith:)`
`ManagedBufferPointer.init(bufferClass:minimumCapacity:initialValue)`  => `init(bufferClass:minimumCapacity:makingHeaderWith:)`

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

A smoke test on macOS does the following:
1. Builds the compiler incrementally.
2. Builds the standard library only for macOS. Simulator standard libraries and
   device standard libraries are not built.
3. lldb is not built.
4. The test and validation-test targets are run only for macOS. The optimized
   version of these tests are not run.

A smoke test on Linux does the following:
1. Builds the compiler incrementally.
2. Builds the standard library incrementally.
3. lldb is built incrementally.
4. The swift test and validation-test targets are run. The optimized version of these
   tests are not run.
5. lldb is tested.

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@dabrahams  Could you take a look?

File: test/1_stdlib/Renames.swift [hunk 1]
```diff
@@ -281,8 +305,14 @@ func _LifetimeManager<T>(x: T) {
   _ = withUnsafePointers(&x, &x, &x) { _, _, _ in } // expected-error {{'withUnsafePointers' is unavailable: use nested withUnsafePointer(to:_:) instead}} {{none}}
 }
 
-func _ManagedBuffer<V, E>(x: ManagedBufferPointer<V, E>) {
+func _ManagedBuffer<H, E>(x: ManagedBufferPointer<H, E>, h: H, bc: AnyClass) {
   _ = x.allocatedElementCount // expected-error {{'allocatedElementCount' has been renamed to 'capacity'}} {{9-30=capacity}} {{none}}
+  _ = ManagedBuffer<H, E>.create(1) { _ in h } // expected-error {{'create(_:initialValue:)' has been renamed to 'create(minimumCapacity:makingHeaderWith:)'}} {{27-33=create}} {{34-34=minimumCapacity: }} {{none}}
+  _ = ManagedBuffer<H, E>.create(1, initialValue: { _ in h }) // expected-error {{'create(_:initialValue:)' has been renamed to 'create(minimumCapacity:makingHeaderWith:)'}} {{27-33=create}} {{34-34=minimumCapacity: }} {{37-49=makingHeaderWith}}i {{none}}
+  _ = ManagedBufferPointer<H, E>(bufferClass: bc, minimumCapacity: 1, initialValue: { _, _ in h }) // expected-error {{'init(bufferClass:minimumCapacity:initialValue:)' has been renamed to 'init(bufferClass:minimumCapacity:makingHeaderWith:)'}} FIXME: fix-it {{none}}
```
Comment (rintaro) [inline test/1_stdlib/Renames.swift]: Add fix-it if #4013 landed.

Comment (CodaFi) [inline test/1_stdlib/Renames.swift]: I just landed this patch.  Please rebase and retest.

Comment (rintaro) [inline test/1_stdlib/Renames.swift]: @CodaFi Thank you!

**Comment by rintaro** [comment]
@swift-ci Please smoke test OS X Platform

**Comment by rintaro** [comment]
Hm status reporting from CI is still broken?
https://ci.swift.org/job/swift-PR-osx-smoke-test/1171/

**Comment by CodaFi** [comment]
It does this sometimes.  Usually another build will fix it.

@swift-ci please smoke test OS X platform.

**Comment by tkremenek** [comment]
@rintaro Looks great!  Please create another pull request for `swift-3.0-branch`.

**Pull request merged**